src/main/java/org/broadinstitute/hellbender/utils/samples/MendelianViolation.java

```diff
@@ -49,14 +57,32 @@ public int getParentsRefRefChildHet() {
         return inheritance.get(GenotypeType.HOM_REF).get(GenotypeType.HOM_REF).get(GenotypeType.HET);
     }
 
-    public boolean isViolation(final Sample mother, final Sample father, final Sample child, final VariantContext vc){
-        violations_total=0;
+    private void resetCounts() {
+        noCall = 0;
+        lowQual = 0;
+        familyCalled = 0;
+        varFamilyCalled = 0;
+        violations_total = 0;
+        violationFamilies.clear();
+
         clearInheritanceMap();
+    }
+
+    /**
+     * Tests whether there is a mendelian violation between the supplied samples.  Note: this will reset any accumulated stats.
+     * @param mother
+     * @param father
+     * @param child
+     * @param vc
+     * @return Whether a violation is present
+     */
+    public boolean isViolation(final Sample mother, final Sample father, final Sample child, final VariantContext vc){
```

Comment:
Having a predicate method like `isViolation()` alter object state like this is a pretty dangerous design pattern. Is there a simple refactoring we could do in this `MendelianViolation` class to fix this? What if we introduced an inner `MendelianViolationCounts` class, stored an instance of it for use by methods that need to accumulate counts across calls, and create a fresh, isolated instance of `MendelianViolationCounts` for methods that currently reset the counts on every call. What do you think @cmnbroad ?

Reply:
I dont disagree at all here, but as @cmnbroad noted, this entire pattern existing in GATK4 before this PR.  Something like a completely separate MendelianViolationCounter class and MendelianViolationUtils (the static methods) isnt a huge refactor, but I'd really prefer to not creep the scope of this PR here.